Dev-Corner/slides/tool_integration.html

```diff
@@ -59,103 +59,1315 @@
+                                        <not_has_text text="chr8" />
+                                        <has_text_matching expression="1274\d+53" />
+                                        <has_line_matching expression=".*\s+127489808\s+127494553" />
+                                        <!-- &#009; is XML escape code for tab -->
+                                        <has_line line="chr7&#009;127471195&#009;127489808" />
+                                        <has_n_columns n="3" />
+                                    </assert_contents>
+                                </output>
+                            </test>
+                            ```
+
+                            <small>[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test)</small>
+
+                        </script>
+                    </section>
+
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### `tests`
+                            #### Comparison Methods on the output itself
+
+                            ```xml
+                            <test>
+                                &lt;output ...
+                                    md5
+                                    compare
+                                        diff
+                                        sim_size
+                                    lines_diff
+                                ...
+
+                            ```
+
+                            <small>[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test|output)</small>
+
+                        </script>
+                    </section>
+
+                    <section data-markdown>
+                        <script type="text/template">
+                          ### `tests`
+                          #### Comparison Methods within the output
+
+                          ```xml
+                          <test>
+                              <output>
+                                  <assert_contents>
+                                      <has_text />
+                                      <not_has_text />
+                                      <has_text_matching />
+                                      <has_text />
+                                      <...>
+                          ```
+                          <small>[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test|output|assert_contents)</small>
+
+                        </script>
+                    </section>
+
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Wrapper directory tree
+
+                            ```
+                            graphlan/
+                            ├── graphlan.xml
+                            ├── wrapper.py
```

Comment:
Even if it's a fake. It is to show that sometime, you will have a wrapper.py between your xml and the executable